src/JSON/Utils/Cursors.dfy

```diff
@@ -142,6 +142,15 @@ module {:options "-functionSyntax:4"} JSON.Utils.Cursors {
     function Split() : (sp: Split<View>) requires Valid?
       ensures sp.SplitFrom?(this, (v: View) => v.Bytes())
       ensures beg != point ==> sp.StrictlySplitFrom?(this, (v: View) => v.Bytes())
+      ensures !BOF? ==> (sp.StrictlySplitFrom?(this, (v: View) => v.Bytes()) && sp.cs.StrictSuffixOf?(this))
```

Comment:
Given how much you're adding to the postconditions, would it be feasible to make `Split` opaque, as well?

Reply:
Removed most ensures clauses.